### PR TITLE
Kafka producer retries

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Bump Kafka producer retries from 5x to 10x
- Actually kill the server if no event ingested in 5min if the `STALENESS_RESTART_SECONDS` env is set. Should help with getting paged.
- Related: https://github.com/PostHog/plugin-server/issues/348

## Checklist

-   [ ] Updated Settings section in README.md, if settings are affected
-   [ ] Jest tests

### CODE REVIEW

File: src/main/pluginsServer.ts [hunk 1]
```diff
@@ -176,17 +176,21 @@ export async function startPluginsServer(
...
                         }
                     )
                     console.log(
-                        `Plugin Server has not ingested events for over ${serverConfig.STALENESS_RESTART_SECONDS} seconds!`,
+                        `Plugin Server has not ingested events for over ${serverConfig.STALENESS_RESTART_SECONDS} seconds! Rebooting.`,
                         extra
                     )
+                    setTimeout(() => process.kill(process.pid, 'SIGINT'), 1000)
+                    setTimeout(() => process.kill(process.pid, 'SIGTERM'), 60000)
+                    setTimeout(() => process.kill(process.pid, 'SIGKILL'), 120000)
```
Comment (Twixes) [inline src/main/pluginsServer.ts:193]: We should log this shutdown to StatsD. BTW isn't SIGINT the same as SIGTERM here?

Comment (mariusandra) [reply src/main/pluginsServer.ts]: Sure, changed to:

```js
                    server.statsd?.increment(`alerts.stale_plugin_server_restarted`)
                    setTimeout(() => process.kill(process.pid, 'SIGTERM'), 1000)
                    setTimeout(() => process.kill(process.pid, 'SIGTERM'), 60000)
                    setTimeout(() => process.kill(process.pid, 'SIGKILL'), 120000)
```

No need to mess around, we're sending a clear signal that it's time to terminate 🕶️

**Comment by mariusandra** [comment]
The only problem now is that the staleness test terminates jest :D

**Comment by mariusandra** [comment]
Should be fixed. 🤞 tests...

I also fixed another leaky timeout. Might as well go in with the same PR...

**Comment by mariusandra** [comment]
@Twixes mind giving it a quick review when you get the change?

**Review by Twixes — APPROVED** [review]
LGTM

**Pull request merged**